pkg/registration/register.go

```diff
@@ -1,18 +1,73 @@
 package registration
 
 import (
-	"github.com/SUSE/connect-ng/internal/collectors"
+	"encoding/json"
+
 	"github.com/SUSE/connect-ng/pkg/connection"
 )
 
+type announceRequest struct {
+	Hostname          string `json:"hostname"`
+	SystemInformation any    `json:"hwinfo"`
+}
+
+type announceResponse struct {
+	Id       int    `json:"id"`
+	Login    string `json:"login"`
+	Password string `json:"password"`
+}
+
 // Register a system by using the given regcode. You also need to provide the
 // hostname of the system plus extra info that is to be bundled when registering
 // a system.
-func Register(conn connection.Connection, regcode, hostname string, info collectors.Result) error {
-	return nil
+func Register(conn connection.Connection, regcode, hostname string, systemInformation any) (int, error) {
```

Comment:
I generalized `info collectors.Result` into `systemInformation any` to remove the internal dependency and also let library users decide which type of information they want to send.

You disagreed?

Reply:
See above comment 😄